### DISCUSSION TITLE
`-l` argument does not work when engine is fofa

### DISCUSSION THREAD

**Discussion by leveryd** [projectdiscovery/uncover#206] [Q&A] [answered]
### uncover version:

1.0.1

### Current Behavior:

```
/ # uncover -q 'cert="baidu" && (icon_hash="-1374555452" || icon_hash="1684500512")' -l 2000 -json -o /tmp/result.json -e fofa
/ # wc -l /tmp/result.json
91 /tmp/result.json
```

`-l 2000` return little record, more fewer than 2000.

besides, the search result is bigger than 2000.

<img width="861" alt="image" src="https://user-images.githubusercontent.com/1846319/226158766-59b30c3e-c89a-49f1-ba80-cbc5cabd9036.png">

### Expected Behavior:

`-l 2000` return 2000 result

### Steps To Reproduce:

```
/ # uncover -q 'cert="baidu" && (icon_hash="-1374555452" || icon_hash="1684500512")' -l 2000 -json -o /tmp/result.json -e fofa
```

**Comment by xm1k3** [comment]
Hi @leveryd, can I ask if you have enough F points in your Fofa profile to complete the request? 
Could it be that you get 91 records back just because there aren't enough F points available for the request?

![image](https://user-images.githubusercontent.com/73166077/230078677-e680fa6c-6c39-46f1-91cb-d9816c2966ef.png)

**Comment by leveryd** [comment]
It is enough.

My account can get 10000 record. when i need to fetch more than 10000 record, it will need F points.

**Comment by bongobongoland** [comment]
I can confirm the bug. Instead of saving 2000 results, only 67 results are saved.
[result.json.txt](https://github.com/projectdiscovery/uncover/files/11163555/result.json.txt)
However, if I run uncover with -v, in the terminal, I can see 2200 results (not 2000 as requested).
[terminal_fofa_baidu_glitch.txt](https://github.com/projectdiscovery/uncover/files/11163576/terminal_fofa_baidu_glitch.txt)

**Comment by ehsandeep** [comment, marked as answer]
@leveryd @bongobongoland, it works for me with latest version!

```console
uncover -q 'cert="baidu" && (icon_hash="-1374555452" || icon_hash="1684500512")' -l 2000 -e fofa | wc -l

  __  ______  _________ _   _____  _____
 / / / / __ \/ ___/ __ \ | / / _ \/ ___/
/ /_/ / / / / /__/ /_/ / |/ /  __/ /    
\__,_/_/ /_/\___/\____/|___/\___/_/

		projectdiscovery.io

[INF] Current uncover version v1.0.4 (latest)
    2200
```

**Comment by bongobongoland** [comment]
Yes, the bug has been fixed. Thanks.
